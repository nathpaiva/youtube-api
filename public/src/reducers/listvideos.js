'user strict';

import { List } from 'immutable';

export function listVideosReducer(state = new List(), action) {

  if (action.type === 'LIST') {
    return new List(action.list);
  }

  return state;
}
