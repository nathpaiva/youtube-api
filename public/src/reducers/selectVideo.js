'user strict';

import { List } from 'immutable';

export function selectVideoReducer(state = new List(), action) {

  if (action.type === 'SELECTFIRST') {
    return [action.list[0]];
  }

  if (action.type === 'SELECTONE') {
    return [action.list];
  }


  return state;
}
