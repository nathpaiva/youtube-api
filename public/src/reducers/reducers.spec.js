'use stict';

import React from 'react';
import renderer from 'react-test-renderer';
import { listVideosReducer } from './listvideos';
import { selectVideoReducer } from './selectVideo';


describe('#Test reduces', () => {
  it('#Should check listvideos', () => {
    const response = listVideosReducer([], { type: 'LIST', list: [] });

    expect(typeof response === 'object').toEqual(true);
    expect(response).toMatchSnapshot();
  });

  it('#Should check selectVideoReducer SELECTFIRST', () => {
    const response = selectVideoReducer([], { type: 'SELECTFIRST', list: [{ id: 0 }, { id: 1 }] });

    expect(response[0].id === 0).toEqual(true);
    expect(response).toMatchSnapshot();
  });

  it('#Should check selectVideoReducer SELECTONE', () => {
    const response = selectVideoReducer([], { type: 'SELECTONE', list: { id: 0 } });
    const isArrayResult = Array.isArray(response);

    expect(isArrayResult).toEqual(true);
    expect(response).toMatchSnapshot();
  });
});
