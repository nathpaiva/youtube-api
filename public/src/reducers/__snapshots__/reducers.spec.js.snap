// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Test reduces #Should check listvideos 1`] = `
Immutable.List [
]
`;

exports[`#Test reduces #Should check selectVideoReducer SELECTFIRST 1`] = `
Array [
  Object {
    "id": 0,
  },
]
`;

exports[`#Test reduces #Should check selectVideoReducer SELECTONE 1`] = `
Array [
  Object {
    "id": 0,
  },
]
`;
